dynamiqs/qarrays/qarray.py

```diff
@@ -66,10 +66,61 @@ def _dims_from_qutip(dims: list[list[int]]) -> tuple[int, ...]:
 
 
 class QArray(eqx.Module):
-    r"""Quantum array object. DenseQArray is a wrapper around JAX arrays. It offers
-    convenience methods to handle more easily quantum states
-    (bras, kets and density matrices) and quantum operators.
-    If you come from QuTiP, this is the equivalent of the Qobj class.
+    r"""Dynamiqs custom array to represent quantum objects.
+
+    A qarray is a wrapper around the data structure representing a quantum object (a
+    ket, a density matrix, an operator, a superoperator, etc.) that offers convenience
+    methods.
+
+    There are two types of qarrays:
+
+    - `DenseQArray`: wrapper around JAX arrays, dense representation of the array.
+    - `SparseDIAQArray`: Dynamiqs sparse diagonal format, storing only the non-zero
+        diagonals.
+
+    Use the constructor [`dq.asqarray()`][dynamiqs.asqarray] to build a qarray.
+
+    Attributes:
+        dtype _(numpy dtype)_: Data type.
+        shape _(tuple of ints)_: Array shape.
+        ndim _(int)_: Number of dimensions of the array.
+        layout _(Layout)_: Data layout, either `dq.dense` or `dq.dia`.
+        dims _(tuple of ints)_: Hilbert space dimension of each subsystem.
+        mT _(QArray)_: Returns the qarray transposed over its last two dimensions.
+
+    Note: Arithmetic operation support
+        Qarrays support elementary operations, such as element-wise
+        addition/subtraction, element-wise multiplication and matrix multiplications
+        with other qarray-like objects.
+
+    Note-: Shortcuts methods to use quantum utilities
```

Comment:
I think this note and its table are quite important (equally to the rest of the docstring IMO). I would propose not having it as a note. This is what PyTorch does for instance, [here](https://pytorch.org/docs/stable/tensors.html#torch.Tensor.mH).

Actually, I would even go further and provide a table for the _all_ methods of a `dq.QArray`, such that:

- for methods that are duplicates of quantum utilities (e.g. `.dag()`), just have a description saying "See `dq.this`" or "Alias of `dq.this`" (like pytorch does)
- for methods that are new to QArrays (e.g. `.conj()` or `reshape(*shape)`), have a short one-liner description + link to the full docstring just below in the page.